rewrite-java-core/src/main/java/com/azure/recipes/v2/TypeReferenceRecipe.java

```diff
@@ -47,15 +47,30 @@ private static class ConvertTypeReferenceVisitor extends JavaIsoVisitor<Executio
                     .anyMatch(methodDeclaration -> methodDeclaration.getName().getSimpleName().equals("getRawType"));
 
             if (!alreadyTransformed && visitedNewClass.toString().contains("TypeReference")) {
-                // Extract type from List type in TypeReference declaration
-                Pattern pattern = Pattern.compile("List<(\\w+)>");
+                // Extract type from generic type in TypeReference declaration
+                Pattern pattern = Pattern.compile("<([\\w.]+)<([\\w.]+)(?:,\\s*([\\w.]+))?>>");
```

Comment:
Instead of parsing the string representation of the class, use the structured metadata provided by `J.NewClass`
here https://github.com/openrewrite/rewrite/blob/main/rewrite-java/src/main/java/org/openrewrite/java/tree/J.java
This will help in giving you access to its type arguments, same way without the regex.

Reply:
Addressed in latest commits.